src/main/java/ch/uzh/ifi/hase/soprafs21/objects/GameEngine.java

```diff
@@ -326,4 +328,33 @@ public Player findPlayerbyUsername(Game game, String playerName){
         }
         return null;
     }
+
+    public void createGameFromGameSessionAndFillUp(GameSession gameSession){
+
+        int numberOfUsersToInvite = 4 - gameSession.getUserList().size();
+
+        if(!gameSession.getInvitedUsers().isEmpty()){
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Cant use FillUp while there are still pending gameRequests");
+        }
+        else if(numberOfUsersToInvite > waitingRoom.getUserCount()){
```

Comment:
Alternative: Wenn mit waiting room usern ≠ 4 players, einfach alle vom waiting room in die game session als akzeptiert reinnehmen?

Reply:
habe ich momentan so gemacht, weil ich damit nicht von edouards implementierung abhängig war.  der ganze prozess von leute werden einer gamesession hinzugefügt und aktuelle userliste des waitingrooms wird gebroadcastet ist ja edouards sache, ich habe aber bisher nichts von ihm gehört, weiss nicht wo er steht damit und ich wollte dies sache um ehrlich zu sein einfach erledigt haben, aber ja es war auch meine idee zuerst es soo zu machen. könnten wir ja mal anschauen, sobald edouard seine funktionalitäten eingebaut hat und seinen branch gepublished oder sogar auf den dev gepusht hat, was meinst du?

Reply:
ok, dann machen wir das mal so und warten auf edouard